### DISCUSSION TITLE
Easier way to access headers?

### DISCUSSION THREAD

**Discussion by lindeberg** [RicoSuter/NSwag#4556] [Q&A]
## Background
`SwaggerResponse.Headers` is  `IReadOnlyDictionary<string,IEnumerable<string>>`. 

## Problem
Accessing headers from from `SwaggerResponse` is really unergonomical, compared to `HttpResponseMessage`.

**To get Cache-Control MaxAge in `HttpResponseMessage`**:
`httpResponseMessage.Headers.CacheControl.MaxAge`

**To get Cache-Control MaxAge in 'SwaggerResponse' (hoping it works and a dozen unit tests)**:

```
    public static int? GetMaxAge(this IReadOnlyDictionary<string, IEnumerable<string>> headers)
    {
        if (!headers.TryGetValue(HeaderNames.CacheControl, out var cacheControlHeaderValues))
        {
            return null;
        }

        var cacheControlValue = cacheControlHeaderValues.FirstOrDefault();

        if (string.IsNullOrWhiteSpace(cacheControlValue))
        {
            return null;
        }

        var headerParts = cacheControlValue.Split(',');

        foreach (var part in headerParts)
        {
            if (!part.Trim().StartsWith("max-age"))
            {
                continue;
            }

            var keyValue = part.Split('=');
            if (keyValue.Length != 2)
            {
                continue;
            }

            return int.TryParse(keyValue[1].Trim(), out var maxAge) ? maxAge : null;
        }

        return null;
    }
```

## Question
Is there an easier way? Or should I open a feature request?

**Comment by mikeronayne** [comment]
`CacheControlHeaderValue.Parse` might be of use to you: https://learn.microsoft.com/en-us/dotnet/api/system.net.http.headers.cachecontrolheadervalue.parse
